sql/core/src/test/scala/org/apache/spark/sql/execution/QueryExecutionSuite.scala

```diff
@@ -336,6 +336,19 @@ class QueryExecutionSuite extends SharedSparkSession {
     }
   }
 
+  test("SPARK-47033: EXECUTE IMMEDIATE USING does not recognize session variable names") {
+    spark.sql("DECLARE parm = 'Hello';")
+
+    val originalQuery = spark.sql(
+      "EXECUTE IMMEDIATE 'SELECT :parm' USING system.session.parm AS parm;")
+    val newQuery = spark.sql("EXECUTE IMMEDIATE 'SELECT :parm' USING system.session.parm;")
+
+    val columns = originalQuery.columns zip newQuery.columns
+    assert(columns.count(x => x._1 != x._2) == 0)
```

Comment:
we can just do `assert(originalQuery.columns == newQuery.columns)`

Reply:
Done.